### PR TITLE
Add Mocked endpoint with OAS examples

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
Fixes https://github.com/wso2/product-apim/issues/1225
A new radio button set has been added as below. 
Also, the expansion panel is updated similar to the operation view on resource pages. 
Created new common components and formatted the existing code.
Screenshot from 2022-02-09 11-57-40

### CODE REVIEW

**Review by tmkasun — APPROVED** [review]

File: portals/publisher/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx [hunk 1]
```diff
@@ -622,8 +625,17 @@ function EndpointOverview(props) {
                     )}
                 </Grid>
                 <Grid item xs={12}>
-                    {endpointType.key === 'INLINE' ? iff(Object.keys(swaggerDef.paths).length !== 0,
-                        <InlineEndpoints paths={swaggerDef.paths} updatePaths={updatePaths} />, <Progress />)
+                    {(endpointType.key === 'INLINE' || endpointType.key === 'MOCKED_OAS') ? 
+                        iff(Object.keys(swaggerDef.paths).length !== 0, 
+                            <MockImplEndpoints 
+                                paths={swaggerDef.paths}
```
Comment (tmkasun) [inline portals/publisher/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx:631]: should use `key` prop here

File: portals/publisher/source/src/app/components/Apis/Details/Resources/components/GenericOperation.jsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+ * @returns {React.Component} @inheritdoc
+ */
+function GenericOperation(props) {
+    const {
+        target,
+        verb,
+        children,
+    } = props;
+    const [isExpanded, setIsExpanded] = useState(false);
+    const useStyles = makeStyles((theme) => {
```
Comment (tmkasun) [inline portals/publisher/source/src/app/components/Apis/Details/Resources/components/GenericOperation.jsx:48]: please move this out of component scope

File: portals/publisher/source/src/app/components/Apis/Details/Endpoints/Prototype/MockedOASOperation.jsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+                                }
+                            </TextField>}
+                        <Typography variant='body1' gutterBottom>
+                            <FormattedMessage
+                                id='Apis.Details.Endpoints.Prototype.MockedOAS.Response.Example'
+                                defaultMessage='Example Value'
+                            />
+                        </Typography>
+                        <Suspense fallback={<CircularProgress />}>
+                            <MonacoEditor
```
Comment (tmkasun) [inline portals/publisher/source/src/app/components/Apis/Details/Endpoints/Prototype/MockedOASOperation.jsx:185]: @ashera96 is adding a new Monaco editor package via the gateway policy feature, Once it's merged shall we switch to that package,

Comment (AmaliMatharaarachchi) [reply portals/publisher/source/src/app/components/Apis/Details/Endpoints/Prototype/MockedOASOperation.jsx:212]: ack

**Pull request merged**